msg/sensor_baro.msg

```diff
@@ -8,3 +8,5 @@ uint32 error_count
 float32 pressure	# static pressure measurement in millibar
 
 float32 temperature	# static temperature measurement in deg Celsius
+
+float32 depth      # depth calculation
```

Comment:
The drivers should only be publishing raw sensor data.

At the moment the architecture is that `sensors/vehicle_air_data` consumes all available `sensor_baro` instances, selects the primary, then uses that to compute pressure altitude, air density, etc that's published as `vehicle_air_data`.

Should we do something similar for the (optional) underwater use case?

Reply:
I now see they have implemented something similar, but just for the MS5837-30BA.  They didn’t include support for the -02BA version but otherwise that code provides similar functionality.

I understand publishing just raw sensor data and I will change the code.  It might be best to separate depth pressure measurements from air pressure measurements.  Either with a flag, a separate field, or an entirely new message.  Although the MS5837 can read atmospheric pressure, it is not particularly accurate.  The spec is +/- 50mBar which I believe is basically worthless in determining altitude.  The MS5837 sensor does need to be calibrated in air (at the surface) before entering the water to provide accurate depth measurements. 

Most boards running PX4 have one or more air pressure sensors included onboard, but for the UUV use case these boards are installed in water proof pressure vessels.  Ideally, these internal barometers could be used for cases where the pressure vessel is pressurized and a change in pressure would indicate a leak.  Mixing air and water pressure measurements doesn’t make sense.

As you mention with the current architecture sensor/vechicle_air_data consumes all the sensor_baro data and publishes the vehicle_air_data.  Do you believe it would be better to separate out water pressure measurements or just overload what’s currently supported (only include multiple of either water or air pressure sensors)?

Reply:
What's the proper name for a barometer underwater? Would `sensor_hydrostatic_pressure` be correct?

> As you mention with the current architecture sensor/vechicle_air_data consumes all the sensor_baro data and publishes the vehicle_air_data. Do you believe it would be better to separate out water pressure measurements or just overload what’s currently supported (only include multiple of either water or air pressure sensors)?

Let's just do it separately and make it easy to enable/disable entirely for the UUV case. It might seem a bit excessive for just a simple depth calculation, but it allows us to funnel in data from any source (sensor, mavlink, uavcan, etc) and apply calibrations or future corrections. I can help do this part quickly if we can settle on the details (naming mostly).

So how about something like `sensor_hydrostatic_pressure` -> `vehicle_depth`?

Reply:
I like the idea of keeping it separate, and I can test it on my UUV.  I’m not completely familiar with the PX4 code base, and I’ll have to investigate how this can be integrated into EKF2.  I believe I’d have to add an option to use this data instead of vehicle_air_data.  There are probably a few other areas like logging, uRTPS message, etc. that would require changes, but if you don’t see any major issues, I can start implementing this. `sensor_hydrostatic_pressure` sounds good to me.